tests/Propel/Tests/Generator/Builder/Om/TableMapBuilderTest.php

```diff
@@ -399,4 +401,48 @@ public function getNormalizedColumnNameMapDefinition(): string
         // check for uniqueness -> unique list has to be the same size
         $this->assertCount(14, array_unique($list));
     }
+
+    /**
+     * @return array
+     */
+    public function stringifyDataProvider(): array
+    {
+        return [
+            [1, 'int should stay int'],
+            [3.14, 'float should stay float'],
+            [null, 'null should stay null'],
+            [true, 'bool should stay bool'],
+            ['literal', 'string should stay string'],
+            ['', 'empty string should stay string'],
+            [' ', 'space should stay string'],
+            ["\n", 'new line should stay string'],
+            ['\'quoted literal\'', 'quotes should be escaped'],
+            [[1, 2, 3], 'array should stay array'],
+            [['nr1' => 1, 'nr2' => 2], 'array indexes should remain'],
+            [[1, 3.14, null, true, 'literal'], 'array types should not change'],
+            [[null, 'nested' => [1, true], [2.71, 'arr' => ['any']]], 'nested arrays should work too'],
+        ];
+    }
+
+    /**
+     * @dataProvider stringifyDataProvider
+     *
+     * @param bool|int|float|string|array|null $input
+     * @param string $message
+     *
+     * @return void
+     */
+    public function testStringify($input, string $message): void
+    {
+        $builder = new class (new Table('any')) extends TableMapBuilder{
+            public function doStringify($value): string
+            {
+                return $this->stringify($value);
+            }
+        };
+        $inputAsString = $builder->doStringify($input);
+        eval("\$output = $inputAsString;");
```

Comment:
Not really sure about the purpose of this `eval` call. And shouldn't there be an escaping backslash before `$inputAsString`. Otherwise, it would be interpolated, I assume.

Reply:
Have you looked at what the test case does? Would you prefer `'$output = ' .  $inputAsString . ';'`?
I prefer it how it is, since the concatenation hurts readability.
Yes, the mixing of stringified php code with actual php code is confusing. Unfortunately, that is the subject of the test case.
Instead of using `eval`, I could validate against stored strings. But I think that is a lesser solution, as it also tests for specific formatting, which might change over time, and really does not matter, as long as the snipped produces correct data.

Reply:
I also think that a concat could suffice here. Since this is a test case, this is fine though IMO.

Reply:
I have changed the variable names, hope that makes things clearer